packages/expo-screen-orientation/ios/ScreenOrientationViewController.swift

```diff
@@ -12,8 +14,18 @@ class ScreenOrientationViewController: UIViewController {
   }
 
   convenience init(defaultScreenOrientationFromPlist: Void) {
-    guard let orientationString = Bundle.main.object(forInfoDictionaryKey: defaultScreenOrientationMask) as? String else {
-      self.init(defaultOrientationMask: .portrait)
+    let allPossibleOrientations: UIInterfaceOrientationMask = doesDeviceHaveNotch ? .allButUpsideDown : .all
+    let ipadSupportedOrientationStrings = Bundle.main.object(forInfoDictionaryKey: ipadSupportedOrientationsKey) as? [String] ?? []
+    let supportedOrientationStrings = (isPad() && !ipadSupportedOrientationStrings.isEmpty) ?
+      ipadSupportedOrientationStrings : Bundle.main.object(forInfoDictionaryKey: supportedOrientationsKey) as? [String] ?? []
+    var supportedInterfaceOrientations: UIInterfaceOrientationMask = ScreenOrientationViewController.getSupportedInterfaceOrientations()
+
+    supportedInterfaceOrientations = supportedInterfaceOrientations.isEmpty ? allPossibleOrientations : supportedInterfaceOrientations
```

Comment:
I would extract this to a separate function as only `supportedInterfaceOrientations` is necessary for the rest of the function (then you can also avoid using `var`).